### DISCUSSION TITLE
Overclocking possible (LS220d)?

### DISCUSSION THREAD

**Discussion by Kulturnilpferd** [1000001101000/Debian_on_Buffalo#115] [Q&A]
Hey there,
is it maybe possible to overclock these slow 800Mhz single cores in a LS220d by modifying the boot image?

**Comment by 1000001101000** [comment]
Good to hear from you.

It might be possible but I have only a vague idea of how that would work.

There are a bunch of references in the kernel to reading the value of the clock from some CPU registers, but that all appears to be reading the value set by the bootrom/bootloader/etc:

https://github.com/torvalds/linux/blob/master/drivers/clk/mvebu/armada-370.c
https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/clock/mvebu-core-clock.txt
https://github.com/torvalds/linux/blob/5bfc75d92efd494db37f5c4c173d3639d4772966/arch/arm/boot/dts/armada-370.dtsi

It might be possible to change those settings from within the bootloader (uboot), searching around I don't see a specific way listed to do it but it does seem like something that could be done by modifying the bootloader code. I don't believe buffalo posts their modified uboot source code, I believe the archive that's out there is just the plain uboot source which would lack some pieces needed to build a working image (if I remember correctly).
